server/src/main/java/org/opensearch/cluster/service/ClusterManagerTaskThrottler.java

```diff
@@ -76,6 +60,21 @@ public ClusterManagerTaskThrottler(
         tasksThreshold = new ConcurrentHashMap<>(128); // setting initial capacity so each task will land in different segment
     }
 
+    /**
+     * To configure more task for throttling, override getClusterManagerThrottlingKey method with task name in task executor.
+     * Verify that throttled tasks would be retry.
+     *
+     * Added retry mechanism in TransportClusterManagerNodeAction so it would be retried for customer generated tasks.
+     *
+     * If tasks are not getting retried then we can register with false flag, so user wont be able to configure threshold limits for it.
+     */
+    protected void registerThrottlingKey(String throttlingKey, boolean retryableOnDataNode) {
+        if (THROTTLING_TASK_KEYS.containsKey(throttlingKey)) {
+            throw new IllegalArgumentException("Duplicate throttling keys are configured ");
```

Comment:
If every key is going to be unique and no two tasks would need to be sharing the same key then do we really need task to pass their own keys

Reply:
Keys will be unique but we want key names to be as per the corresponding tasks. Throttler can't determine from the task as which task it is, so each service will pass the corresponding unique key.

Reply:
The other way to look at it is throttler assigns keys and guarantees it only hands out uniques keys rather than each task registering with its set of keys. So we don't need key names instead it could be classname-uuid auto generated by the throttler

Reply:
We will need register functionality for plugin extensibility. As of now we are not adding support for it in plugin interface, but whenever we add it we can leverage this register functionality for it. 

We will return the throttling key as part of the registration and will use that key via submitting the tasks. That way we can at least enforce that keys are not duplicate and not getting reused across classes.